databases/ClickHouse/Portfile

```
@@ -0,0 +1,90 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github  1.0
+PortGroup           cmake   1.1
+
+github.setup        ClickHouse ClickHouse 22.4.2.1 v -stable
+
+description         ClickHouse® is a free analytics DBMS for big data
+
+long_description \
+    ClickHouse® is an open-source column-oriented database management \
+    system that allows generating analytical data reports in real-time.
+
+notes {
+    Install the ClickHouse-server port to run ClickHouse as a system service.
+}
+
+categories          databases
+maintainers         @slarew openmaintainer
+license             Apache-2
+platforms           darwin
+
+# clickhouse overrides the deployment target so match it
+macosx_deployment_target 10.15
+
+installs_libs       no
+
+# require vanilla clang
+compiler.whitelist  macports-clang-13 macports-clang-14
+
+cmake.build_type    RelWithDebInfo
+cmake.generator     Ninja
+
+variant debug description "Enable debug binaries" {
+    cmake.build_type    Debug
+}
+
+# Fetch with git and then fetch all the submodules.
+fetch.type          git
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init --recursive --depth 1"
+}
+
+# Reset these variables to remove ${prefix}/include because ClickHouse really
+# wants to build against its own dependencies.
+compiler.cpath
+compiler.library_path
+configure.env
+configure.ldflags
+configure.cppflags
```

You probably shouldn't completely clear `configure.ldflags`. It contains flags other than the `-L` flag that we also want. `configure.cppflags` currently only contains the `-I` flag but maybe it will contain other flags in the future so future-proof things by only removing what you actually don't want.
```suggestion
configure.ldflags-delete -L${prefix}/lib
configure.cppflags-delete -I${prefix}/include
```